src/cli/api/local_api.py

```diff
@@ -697,36 +703,46 @@ def get_experiment_rounds(self, path: Path) -> int:
         Returns:
             int: Value of the number of rounds
         """
-        # read experiment.json and check the meta->number_of_rounds attribute
+        experiment_data = utils.read_json_file(path / "experiment.json")
+        if "meta" in experiment_data:
+            if "number_of_rounds" in experiment_data["meta"]:
+                return cast(int, experiment_data["meta"]["number_of_rounds"])
+
         return 1
 
     def delete_experiment(self, path: Path) -> None:
         pass
 
-    def run_experiment(self, path: Path) -> Optional[List[ResultType]]:
-        roundSetManager = RoundSetManager()
-        roundSetManager.prepare_input(path)
-        roundSetManager.process()
-        roundSetManager.terminate()
-        return []
+    def run_experiment(self, path: Path) -> GeneratedResultType:
+        roundSetManager = RoundSetManager(asset=self._get_asset(path), path=path)
+        roundSetManager.prepare_input()
```

Comment:
I would rather see that you keep input_parser the same as the one in ns-backend instead of copying stuff to RoundSetManager because we want to share these files in both applications.
Then prepare_input can be done from process instead of here.

Reply:
Moved input_parser out of RoundSetManager